engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -1272,6 +1274,10 @@ class Compiler(
   def updateMetadata(sourceIr: IRModule, copyOfIr: IRModule): IRModule = {
     passManager.runMetadataUpdate(sourceIr, copyOfIr)
   }
+
+  private def printDiagnostic(message: String): Unit =
+    if (config.isStrictErrors) output.println(message)
+    else context.log(Level.WARNING, message)
```

Comment:
Flooding log with [user errors](https://github.com/orgs/enso-org/discussions/8271#discussioncomment-7529419) is probably not very good idea in the long term.